src/PrestaShopBundle/Form/Admin/Improve/International/Translations/CopyLanguageDataProvider.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\International\Translations;
+
+use PrestaShop\PrestaShop\Core\Addon\AddonRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Addon\Theme\Theme;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+
+/**
+ * Class is responsible of managing the data manipulated using copy language form
+ * in "Improve > International > Translations" page.
+ */
+final class CopyLanguageDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var AddonRepositoryInterface
+     */
+    private $themeRepository;
+
+    /**
+     * @param AddonRepositoryInterface $themeRepository
+     */
+    public function __construct(AddonRepositoryInterface $themeRepository)
+    {
+        $this->themeRepository = $themeRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setData(array $data)
+    {
+        return $this->validate($data);
```

Comment:
this seems like a lie. 😄 `CopyLanguageDataProvider ` is data provider, but it does not return data & does not save it. For me, it seems more like validation service than data provider, wdyt?

Reply:
agree :-)